### PR TITLE
Fix table width and code block line returns

### LINKED ISSUE #15: Nested `<code>` tags lead to unsightly line breaks
## Describe the bug

See [arcaflow-plugin-horreum-client/#1](https://github.com/arcalot/arcaflow-plugin-horreum-client/pull/1#discussion_r1415658333).

## To reproduce

See [arcaflow-plugin-horreum-client/#1](https://github.com/arcalot/arcaflow-plugin-horreum-client/pull/1#discussion_r1415658333).

## Additional context

See [arcaflow-plugin-horreum-client/#1](https://github.com/arcalot/arcaflow-plugin-horreum-client/pull/1#discussion_r1415658333).  😉

### PR DESCRIPTION
**Pull request by dustinblack**
## Changes introduced with this PR

Fixes #15 and #26 

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by mfleader — APPROVED** [review]

File: templates/markdown/partials/enum_string.md.tpl [hunk 1]
```diff
@@ -4,9 +4,9 @@
             <ul>
                 {{- range $value, $display := .ValidValues -}}
                     {{- if $display -}}
-                        <li><strong><code>{{ $value }}</code>:</strong> {{ $display.Name }}</li>
+                        <li><strong><code style="white-space: nowrap">{{ $value }}</code>:</strong> {{ $display.Name }}</li>
                     {{- else -}}
-                        <li><strong><code>{{ $value }}</code></strong></li>
+                        <li><strong><code style="white-space: nowrap">{{ $value }}</code></strong></li>
                     {{- end -}}
```
Comment (webbnh) [inline templates/markdown/partials/enum_string.md.tpl:10]: Consider DRYing this:
```
                    <li><strong><code style="white-space: nowrap">{{ $value }}</code></strong>{{- if $display -}}: {{ $display.Name }}{{- end -}}</li>
```
or perhaps
```
                    <li><strong><code style="white-space: nowrap">{{ $value }}</code></strong>
                    {{- if $display -}}<strong>:</strong> {{ $display.Name }}{{- end -}}
                    </li>
```

File: .github/workflows/reusable_workflow.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ jobs:
 
       - name: Install arcaflow-docsgen
         env:
-          DOCSGEN_VER: "0.3.0"
+          DOCSGEN_VER: "0.3.1"
```
Comment (webbnh) [inline .github/workflows/reusable_workflow.yaml:19]: This should really be an input variable -- it shouldn't be defined here, such that we have to keep updating it.

(However, perhaps this is out of scope for this PR....)

File: templates/markdown/partials/property.md.tpl [hunk 3]
```diff
@@ -33,5 +33,5 @@
     {{ . -}}
 {{- end }}</td></tr>{{ end -}}
 {{- with .Examples }}<tr><th>Examples (JSON encoded):</th><td>
-{{ range . }}<pre><code>{{ . }}</code></pre>{{ end }}
+{{ range . }}<pre><code style="white-space: nowrap">{{ . }}</code></pre>{{ end }}
 </td></tr>{{ end -}}
```
Comment (webbnh) [inline templates/markdown/partials/property.md.tpl]: This file is missing its trailing newline.  Ditto `templates/markdown/partials/string.md.tpl`.

**Review by webbnh — APPROVED** [review]
Thanks @dustinblack for knocking this off.  (I started looking into some of these wrinkles some time back but got lost in the layers.)

I have a couple of nits and a suggestion, as well as a discussion point, for consideration.

**Review by dbutenhof — APPROVED** [review]
I'll accept that this probably works, but I'm curious exactly what makes strings wrap. Is that a side-effect of the 100% table width? I suppose that's implied by adding the no-wrap directives for the code styling...

**Comment by webbnh** [comment]
> I'll accept that this probably works, but I'm curious exactly what makes strings wrap. Is that a side-effect of the 100% table width? I suppose that's implied by adding the no-wrap directives for the code styling...

I believe the default for `width` is `auto`; so overriding that and setting it to `100%` instead is limiting the width to some previous constraint (forcing the text to wrap) and preventing that constraint from being relaxed.

File: templates/markdown/partials/enum_string.md.tpl (hunk 1 above)
Comment (dustinblack) [reply templates/markdown/partials/enum_string.md.tpl:10]: I'm happy to implement this based on your code recommendation, but I'm not very familiar with the templating here and was only tackling the HTML/CSS portions that I can test locally and feel confident about.

File: .github/workflows/reusable_workflow.yaml (hunk 2 above)
Comment (dustinblack) [reply .github/workflows/reusable_workflow.yaml:19]: Let me see if there is a quick fix for this.

**Comment by dustinblack** [comment]
> > I'll accept that this probably works, but I'm curious exactly what makes strings wrap. Is that a side-effect of the 100% table width? I suppose that's implied by adding the no-wrap directives for the code styling...
> 
> I believe the default for `width` is `auto`; so overriding that and setting it to `100%` instead is limiting the width to some previous constraint (forcing the text to wrap) and preventing that constraint from being relaxed.

My understanding is that the table width in this case will default to the width of its content, which in this case includes a string with no explicit breaks. Setting the table width to 100% makes its width subordinate to the width of the parent element (in this case, another table).

It's probably better to fix this with a global CSS entry, but we are only rendering sub-elements of a page here, and applying the width here to the element in the template was the simplest approach to fix the rendering issue. I did test these changes directly in the Chrome developer tools, and things rendered as I expected.

File: .github/workflows/reusable_workflow.yaml (hunk 2 above)
Comment (dustinblack) [reply .github/workflows/reusable_workflow.yaml:19]: I can't find a quick solution for this. The reusable workflow is going to run in the context of the calling repo, so `${{ github.ref_name }}` is going to return the version of the calling repo, not the version of the called repo (arcaflow-docsgen). If we change it to an input variable, then we need to change every call to the reusable workflow to pass that variable, and we have to maintain that at each point, which would be a drift maintenance nightmare. There also doesn't seem to be a built-in way to refer to the latest release within a workflow like this.

File: templates/markdown/partials/enum_string.md.tpl (hunk 1 above)
Comment (webbnh) [reply templates/markdown/partials/enum_string.md.tpl:10]: Understood.  I'm not very familiar with this particular templating system, either, but I've worked with other templating systems which are similar, and figured it was easy enough for you to just try it.  😇

The challenging part here is the colon and whether we need/want it to be in "strong" face font; otherwise, it's pretty straightforward.

BTW, thanks for tackling the CSS stuff...I've managed to avoid learning much of that, so far.  😉
